server/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -265,28 +267,7 @@ public final String getNodeName() {
         ClusterStateTaskExecutor<T> executor,
         ClusterStateTaskListener listener
     ) {
-        submitStateUpdateTasks(source, Collections.singletonMap(task, listener), config, executor);
+        masterService.submitStateUpdateTasks(source, Collections.singletonMap(task, listener), config, executor);
     }
 
-    /**
-     * Submits a batch of cluster state update tasks; submitted updates are guaranteed to be processed together,
-     * potentially with more tasks of the same executor.
-     *
-     * @param source   the source of the cluster state update task
-     * @param tasks    a map of update tasks and their corresponding listeners
-     * @param config   the cluster state update task configuration
-     * @param executor the cluster state update task executor; tasks
-     *                 that share the same executor will be executed
-     *                 batches on this executor
-     * @param <T>      the type of the cluster state update task state
-     *
-     */
-    public <T> void submitStateUpdateTasks(
```

Comment:
I wonder if we would want to keep this or if there is reasons for this not to be used. submitting a single executor with multiple tasks sounds like a good thing, like the base primitive.

Reply:
> submitting a single executor with multiple tasks sounds like a good thing, like the base primitive

I guess it's not used because we rely on the batching in the master service and don't do our own batching upstream. I guess the reason it's not used is that if you had to submit a known list of tasks, you might as well just submit a single task doing all the work in one go?
I'm having a hard time seeing a use case for this where I'd want to submit a list of tasks instead of just building a one of task myself.